### PR TITLE
add endpoint handler base class

### PR DESCRIPTION
**Pull request by richardaspinall**
### Description
This PR makes adding endpoint handlers consistent. 

- A new base class "BaseHandler" is extended from any ConcreteHandler. 
- A route config file makes it easy to add the details of the ConcreteHandler, and is automatically configured from there

### How to test

Run unit tests – everything should work the same

### Task

CU-86cu5p8b7

### Checklist

- [X] Added unit tests
- [X] Added learning objectives to clickup task

### CODE REVIEW

**Comment by richardaspinall** [comment]
Task linked: [CU-86cu5p8b7 add base class for endpoint handlers](https://app.clickup.com/t/86cu5p8b7)

File: backend/src/routes/index.ts
Comment (richardaspinall) [inline backend/src/routes/index.ts:1]: @tommy-trombone this takes care of calling the runHandler method dynamically

File: backend/src/routes/routeConfig.ts
Comment (richardaspinall) [inline backend/src/routes/routeConfig.ts:1]: @tommy-trombone this is where routes are defined, the httpMethod and the appropriate handler

File: backend/src/shared/BaseHandler.ts
Comment (richardaspinall) [inline backend/src/shared/BaseHandler.ts:1]: @tommy-trombone this is the abstract class that ensures we keep the handlers consistent. we will add to this in time (specifically the validation function)

**Review by tommy-trombone — APPROVED** [review]
Looks awesome. Love how abstracted/modular it is.

**Pull request merged**